mlos_bench/mlos_bench/config/environments/apps/redis/redis.jsonc

```diff
@@ -1,87 +1,103 @@
-[
```

Comment:
Reworks environment list configs into CompositeEnv configs as discussed in #340

Reply:
Just to illustrate, here's a hierarchy that we have now:
```
CompositeEnv :: 'Config for: Azure VM Ubuntu Redis'
    VMEnv :: 'Deploy Ubuntu VM on Azure'
    CompositeEnv :: 'Setup Ubuntu VM'
        OSEnv :: 'Boot Ubuntu VM on Azure',
        LocalFileShareEnv :: 'Generate boot parameters for a Ubuntu VM',
        RemoteEnv :: 'Apply boot parameters to a remote Ubuntu VM'
    CompositeEnv :: 'Setup Ubuntu VM'
        LocalFileShareEnv :: 'Generate Linux kernel parameters for Ubuntu'
        RemoteEnv :: 'Apply Linux kernel parameters to a remote Ubuntu VM'
    CompositeEnv :: 'Redis Composite Env'
        LocalFileShareEnv :: 'Generate Redis configuration'
        RemoteEnv :: 'Benchmark Redis on Linux'
        LocalFileShareEnv :: 'Download and post-process Redis benchmark data'
```
Before that, we used to have a single root `CompositeEnv` with a flat list of child environments underneath.

P.S. I am not saying that the new schema is worse or better, just different

Reply:
P.S. I'll probably add a function to print that tree to the log